### DISCUSSION TITLE
QoL Improvement for winget app installation

### DISCUSSION THREAD

**Discussion by bidaumm** [microsoft/winget-cli#3827] [Ideas] [answered]
The suggestion comes from my usage of the ArchLinux [Yay - Aur Helper](https://github.com/Jguer/yay)
Yay allows you to search for and install an app in a single command which is such a nice workflow improvement.
For instance, if I want to install ProtonVPN I can do the following process:

```
> yay install proton
2 aur/protonvpn 1.0.0-3 (+73 6.27) (Out-of-date: 2023-10-10)
    Official ProtonVPN metapackage that installs protonvpn-gui and protonvpn-cli, maintained by the ProtonVPN team.
1 aur/protonup-qt 2.8.2-1 (+43 1.89)
    Install and manage Proton-GE and Luxtorpeda for Steam and Wine-GE for Lutris
==> Packages to install (eg: 1 2 3, 1-3 or ^4)
> 2
```

Currently when installing apps via winget I have to use `winget search` followed by `winget install` and also copy n pasting text from the search command into the install command. 99% of the time I have to do this because I don't know the ID of the app, which can sometimes be different from the application name. Rivatuner is a good example of this:

```
> winget install rivatuner
No package found matching input criteria.
> winget search rivatuner
Name                        Id          Version Source
-------------------------------------------------------
Rivatuner Statistics Server Guru3D.RTSS 7.3.4   winget
> winget install guru3d.rtss
Found Rivatuner Statistics Server [Guru3D.RTSS] Version 7.3.4
```

A huge boon would be to allow a configuration which modifies this process to if `No package found matching input criteria.` is encounter to instead perform the `winget search` function and then if something is returned  provide the list of apps with a reference ID and a input for 'Please select the ID of the package you wish to install'

**Comment by Trenly** [comment]
* Related to https://github.com/microsoft/winget-cli/issues/1620

**Comment by denelon** [comment, marked as answer]
* https://github.com/microsoft/winget-cli/issues/3847

**Comment by 32n1** [comment]
How can i help to bring this thing alive?

**Reply by denelon** [reply]
Are you interested in implementing the feature or trying to raise priority?
